enumerable_custom.rb

```diff
@@ -0,0 +1,126 @@
+# rubocop:disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+def check_all_args(arg)
+  if arg.class == Regexp
+    my_each { |item| return false if arg.match?(item.to_s) == false }
+  elsif arg.class == Class
+    my_each { |item| return false if item.is_a?(arg) == false }
+  else
+    my_all? { |item| item == arg }
+  end
+end
+
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    i = 0
+    array = to_a
+    while i < array.length
+      yield(array[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    array = to_a
+    i = 0
+    until i == array.length
+      yield(array[i], i)
+      i += 1
+    end
+  end
+
+  def my_select(*)
+    return enum_for unless block_given?
+
+    array = []
+    my_each { |item| array << item if yield(item) }
+    array
+  end
+
+  def my_all?(arg = nil)
+    return check_all_args(arg) unless arg.nil?
+
+    condition = true
+    if block_given?
+      my_each { |item| condition = false if yield(item) == false }
+    else
+      my_each { |item| condition = false if item.nil? || item == false }
+    end
+    condition
+  end
```

Comment:
`my_all?` identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2)) 
- when a class is passed as an argument returns true if all of the collection is a member of such class
```ruby
[1, 2, 3].my_all?(Integer) #should return true
[1, 2, nil].my_all?(Integer) #should return false
```
- when a Regex is passed as an argument returns true if all of the collection matches the Regex
```ruby
%w[dog door rod blade].my_all?(/d/)   #should return true.
%w[dog door rod blade].my_all?(/t/)   #should return false.
```